File: src/main/java/core/scm/specialforms/Comment.java
package core.scm.specialforms;

import core.environment.Environment;
import core.evaluator.Evaluator;
import core.scm.Void;

import java.util.List;

/**
 * Comment Special Form:
 *
 * Ignores body and returns nothing
 *
 * Syntax:
 * (comment <expression1> ... <expression n>)
 */
public enum Comment implements ISpecialForm {
  COMMENT;

  @Override
  public Void eval(List<Object> expression, Environment env, Evaluator evaluator) {
    return Void.VOID;
  }

  @Override
  public String toString() {
    return "comment";
  }
}

File: src/main/kotlin/core/scm/specialforms/Comment.kt
package core.scm.specialforms

import core.environment.Environment
import core.evaluator.Evaluator

/**
 * Comment Special Form:
 * Ignores body and returns nothing
 * Syntax:
 * (comment <expression1> ... <expression n>)
 */
object Comment : SpecialForm("comment") {

    override fun eval(form: List<Any?>, env: Environment, evaluator: Evaluator) = Unit
}
